File: OPENAPI-REST-API/openapi-client/scalaz/src/main/scala/org/openapitools/client/api/BackupOrderPutResponse.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import BackupOrderPutResponse._

case class BackupOrderPutResponse (
  continue: Option[Boolean],
errors: Option[List[String]],
serviceType: Option[Integer],
serviceCost: Option[String],
originalCost: Option[String],
repeatServiceCost: Option[String],
hostname: Option[String],
password: Option[String],
coupon: Option[String],
couponCode: Option[Integer])

object BackupOrderPutResponse {
  import DateTimeCodecs._

  implicit val BackupOrderPutResponseCodecJson: CodecJson[BackupOrderPutResponse] = CodecJson.derive[BackupOrderPutResponse]
  implicit val BackupOrderPutResponseDecoder: EntityDecoder[BackupOrderPutResponse] = jsonOf[BackupOrderPutResponse]
  implicit val BackupOrderPutResponseEncoder: EntityEncoder[BackupOrderPutResponse] = jsonEncoderOf[BackupOrderPutResponse]
}

File: OPENAPI-REST-API/openapi-client/zapier/models/BackupOrderPutResponse.js
const utils = require('../utils/utils');

module.exports = {
    fields: (prefix = '', isInput = true, isArrayChild = false) => {
        const {keyPrefix, labelPrefix} = utils.buildKeyAndLabel(prefix, isInput, isArrayChild)
        return [
            {
                key: `${keyPrefix}continue`,
                label: `[${labelPrefix}continue]`,
                type: 'boolean',
            },
            {
                key: `${keyPrefix}errors`,
                label: `[${labelPrefix}errors]`,
                list: true,
                type: 'string',
            },
            {
                key: `${keyPrefix}serviceType`,
                label: `[${labelPrefix}serviceType]`,
                type: 'integer',
            },
            {
                key: `${keyPrefix}serviceCost`,
                label: `[${labelPrefix}serviceCost]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}originalCost`,
                label: `[${labelPrefix}originalCost]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}repeatServiceCost`,
                label: `[${labelPrefix}repeatServiceCost]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}hostname`,
                label: `[${labelPrefix}hostname]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}password`,
                label: `[${labelPrefix}password]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}coupon`,
                label: `[${labelPrefix}coupon]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}couponCode`,
                label: `[${labelPrefix}couponCode]`,
                type: 'integer',
            },
        ]
    },
    mapping: (bundle, prefix = '') => {
        const {keyPrefix} = utils.buildKeyAndLabel(prefix)
        return {
            'continue': bundle.inputData?.[`${keyPrefix}continue`],
            'errors': bundle.inputData?.[`${keyPrefix}errors`],
            'serviceType': bundle.inputData?.[`${keyPrefix}serviceType`],
            'serviceCost': bundle.inputData?.[`${keyPrefix}serviceCost`],
            'originalCost': bundle.inputData?.[`${keyPrefix}originalCost`],
            'repeatServiceCost': bundle.inputData?.[`${keyPrefix}repeatServiceCost`],
            'hostname': bundle.inputData?.[`${keyPrefix}hostname`],
            'password': bundle.inputData?.[`${keyPrefix}password`],
            'coupon': bundle.inputData?.[`${keyPrefix}coupon`],
            'couponCode': bundle.inputData?.[`${keyPrefix}couponCode`],
        }
    },
}
